File: backend/app/schemas/institution.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr, Field


class InstitutionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    legal_name: str
    tax_id: str | None
    email: EmailStr
    phone: str | None
    address: str | None
    city: str | None
    country: str | None
    website: str | None
    logo: str | None
    status: str
    created_at: datetime


class InstitutionUpdate(BaseModel):
    legal_name: str | None = Field(default=None, min_length=3, max_length=255)
    tax_id: str | None = Field(default=None, max_length=50)
    email: EmailStr | None = None
    phone: str | None = Field(default=None, max_length=50)
    address: str | None = None
    city: str | None = Field(default=None, max_length=120)
    country: str | None = Field(default=None, max_length=120)
    website: str | None = Field(default=None, max_length=255)

File: src/types/institution.ts
export type InstitutionStatus = 'ACTIVE' | 'SUSPENDED' | 'PENDING'
export type PlanId = 'starter' | 'professional' | 'enterprise'

export interface Institution {
  id: string
  legalName: string
  taxId: string | null
  email: string
  phone: string | null
  address: string | null
  city: string | null
  country: string | null
  website: string | null
  logo: string | null
  status: InstitutionStatus
  plan: PlanId
  planActivatedAt: string | null
  createdAt: string
  updatedAt: string
}

export interface InstitutionUpdatePayload {
  legalName: string
  taxId?: string | null
  email: string
  phone?: string | null
  address?: string | null
  city?: string | null
  country?: string | null
  website?: string | null
}

export interface DashboardStats {
  totalDocuments: number
  validDocuments: number
  revokedDocuments: number
  totalVerifications: number
  recentActivity: AuditEvent[]
}

export type AuditEventType =
  | 'USER_CREATED'
  | 'LOGIN'
  | 'DOCUMENT_CREATED'
  | 'DOCUMENT_SIGNED'
  | 'DOCUMENT_VERIFIED'
  | 'DOCUMENT_REVOKED'
  | 'INSTITUTION_UPDATED'
  | 'USER_UPDATED'

export interface AuditEvent {
  id: string
  actorId: string
  actorName: string
  action: AuditEventType
  entityType: string
  entityId: string
  metadata: Record<string, unknown> | null
  createdAt: string
}

export interface AuditListParams {
  page?: number
  pageSize?: number
  action?: AuditEventType
  search?: string
}
